compiler/rustc_hir_typeck/src/generator_interior/mod.rs

```diff
@@ -426,6 +445,13 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
                 }
                 _ => intravisit::walk_expr(self, expr),
             },
+            ExprKind::Assign(_, rhs, _) => {
+                if self.task_context_hid.is_some() && matches!(rhs.kind, hir::ExprKind::Yield(..)) {
+                    self.in_await_yield = true;
+                }
+                intravisit::walk_expr(self, expr);
+                self.in_await_yield = false;
+            }
```

Comment:
Doesn't this walk the LHS as if we were inside of a yield too? Is that the intention? If so, `in_await_yield` needs a better name. If not, then it needs fixing I think.

Might be worthwhile to leave a comment explaining what kind of expression you expect to see here, and why.